todo_react/todo-react/src/App.tsx

```diff
@@ -0,0 +1,60 @@
+import React, {useState} from 'react';
+import './App.css';
+import {AddToDo} from "./components/AddToDo";
+import {ToDoItem} from "./components/ToDoItem";
+import {ITodo} from "./models";
+import {TodosContext} from "./context"
+
+export const App: React.FC = () => {
+    let todoList = [...JSON.parse(localStorage.getItem('TodoList') || "")]
+    const [todos, setTodos] = useState<ITodo[]>(todoList)
```

Comment:
`<TodosProvider>` should create `todos` as state, not `<App>`
`<App>` should only take `todos` from context via `useTodosContext` or `useContext(TodosContext)`